Dropwizard Extra ZooKeeper

jairamc opened the pull request:
Added a new module that introduces ZooKeeper integration in Dropwizard.

Commit:
Dropwizard Extra ZooKeeper

Added a new module that introduces ZooKeeper integration in Dropwizard.

nicktelford commented:
Excellent, I'm glad someone else feels a need for this, I've been working on a `dropwizard-extra-zookeeper` module as part of some integration with Netflix's Curator in a branch: https://github.com/datasift/dropwizard-extra/tree/feature/zookeeper-curator

My one suggestion would be to manage the ZooKeeper client instance using composition, rather than inheritance; something like this: https://github.com/datasift/dropwizard-extra/blob/feature/zookeeper-curator/dropwizard-extra-zookeeper/src/main/java/com/datasift/dropwizard/zookeeper/ManagedZooKeeper.java

Otherwise, looks good and I'm more than happy to merge.

nicktelford commented:
BTW: the other features I'm working on are a revised `HealthCheck` for ZooKeeper connections and to fully instrument them with `metrics`. Some of this I've yet to push up to github so you won't be able to see it yet.

Commit:
Changed ZooKeeper initialization to composition

ManagedZooKeeperClient no longer inherits from ZooKeeper class. Instead it only manages an instance. The ZooKeeper instance is created by the ZooKeeperClientFactory class

Commit:
Fixed javadoc for ManagedZookeeperClient

nicktelford commented on dropwizard-extra-zookeeper/src/main/java/com/datasift/dropwizard/ZookeeperClientFactory.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Factory to generate {@link ManagedZooKeeperClient} instances from the config.
+ */
+public class ZookeeperClientFactory {
```
Nitpick: Can you rename this "ZooKeeperClientFactory" so that capitalization of "ZooKeeper" is consistent (and in-line with the underlying library) please?

jairamc replied:
My bad. Missed that one. Will fix.

Commit:
Changed ZookeeperClientFactory to ZooKeeperClientFactory

That's with a capital K to be inline with the underlying library

jairamc commented on dropwizard-extra-zookeeper/src/main/java/com/datasift/dropwizard/ZooKeeperClientFactory.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Factory to generate {@link ManagedZooKeeperClient} instances from the config.
+ */
+public class ZooKeeperClientFactory {
```
The change has been made.
